repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java

```diff
@@ -105,6 +105,32 @@ public List<AtlasTask> getPendingTasks() {
         return ret;
     }
 
+    @GraphTransaction
+    public List<AtlasTask> getInProgressTasks() {
+        List<AtlasTask> ret = new ArrayList<>();
+
+        try {
+            AtlasGraphQuery query = graph.query()
+                    .has(Constants.TASK_TYPE_PROPERTY_KEY, Constants.TASK_TYPE_NAME)
+                    .has(Constants.TASK_STATUS, AtlasTask.Status.IN_PROGRESS)
+                    .orderBy(Constants.TASK_CREATED_TIME, AtlasGraphQuery.SortOrder.ASC);
+
+            Iterator<AtlasVertex> results = query.vertices().iterator();
+
+            while (results.hasNext()) {
+                AtlasVertex vertex = results.next();
+
+                ret.add(toAtlasTask(vertex));
+            }
+        } catch (Exception exception) {
+            LOG.error("Error fetching in progress tasks!", exception);
+        } finally {
+            graph.commit();
```

Comment:
Its method to get in progress task vertices [read call], is `graph.commit()` required here ? Please review.

Reply:
You are correct. I just copied this from `getPendingTasks` in line 83 and updated the status in the query.

Reply:
Should I change the behavior in `getPendingTasks` as well ?

Reply:
> Should I change the behavior in getPendingTasks as well ? 
Yes if is not impacting transaction bebaiovour while processing tasks.

Reply:
addressed